src/components/RequestHistoryPanel.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useEffect, useState } from "react"
+import {
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  Button,
+  Divider,
+  Paper,
+  Stack,
+  Typography,
+} from "@mui/material"
+import Grid2 from "@mui/material/Unstable_Grid2/Grid2"
+import { CodeEditor } from "@/components/CodeEditor"
+import { useActiveAddress } from "@arweave-wallet-kit/react"
+
+const ChevronDownIcon = () => (
+  <svg width="24" height="24" fill="none" viewBox="0 0 24 24" style={{ display: "block" }}>
+    <path
+      stroke="currentColor"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+      strokeWidth="1.5"
+      d="M15.25 10.75L12 14.25L8.75 10.75"
+    />
+  </svg>
+)
+
+const truncateMiddle = (str: string, front = 8, back = 8) => {
+  if (!str || str.length <= front + back) return str
+  return `${str.slice(0, front)}...${str.slice(-back)}`
+}
+
+interface RequestHistoryPanelProps {
+  onSelect: (payload: string) => void
+}
+
+export function RequestHistoryPanel({ onSelect }: RequestHistoryPanelProps) {
+  const address = useActiveAddress()
+  const [history, setHistory] = useState<any[]>([])
+  const [expanded, setExpanded] = useState<string | false>(false)
+  const [hoveredId, setHoveredId] = useState<string | null>(null)
+
+  const loadHistory = () => {
```

Comment:
By the way, you can use `@nanostores/persistent"` to get rid of a bit of boilerplate when saving a json to local storage.

```typescript
import { persistentAtom } from '@nanostores/persistent'

export const shoppingCart = persistentAtom<Product[]>('cart', [], {
  encode: JSON.stringify,
  decode: JSON.parse,
})
```

The package is already installed in the repo.